packages/downloading-helpers/src/scripts/replay-assets.ts

```diff
@@ -44,10 +47,14 @@ export const fetchAsset: (path: string) => Promise<string> = async (path) => {
     if (entry && etag !== "" && etag === entry.etag) {
       logger.debug(`${fetchUrl} already present`);
       releaseLock();
+      await fetchSourceMapIfDebugging(fetchUrl);
       return filePath;
     }
 
-    const contents = (await client.get(fetchUrl)).data;
+    let contents = (await client.get(fetchUrl)).data;
+    if (path.endsWith(".map")) {
+      contents = JSON.stringify(contents);
```

Comment:
The library we use to make the HTTP request is clever enough to realise the body of the response is a JSON object, and it turns it into a JS object rather than a string, which causes errors further down when we try to write out its contents because that code is expecting a string.

Reply:
Ah, that's a bit too clever! I'd either add a comment to explain, or see if Axois has another proprety for raw string data (maybe `.data` is not what we want?)

`if (typeof path !== "string")` could also be more direct: that's what we actually care about.

But I expect Axois `.data` is not the field we want.

Reply:
Unfortunately there isn't a better field! But I agree that checking the type (I believe you mean of `contents` rather than `path`) communicates the intent better. I've also added a comment to explain why we need the type check.

Reply:
You can also avoid the let here too if you like: `await writeFile(filePath, typeof contents !== "string" ? JSON.stringify(contents) : contents);`, but also fine as is.

Reply:
Let's leave this one, I think having a separate line here improves readability a bit and gives us somewhere to put the comment.